apps/consumer-client/pages/examples/add-pcd.tsx

```diff
@@ -62,11 +74,201 @@ export default function Page() {
+
+    const ethereum = (window as any).ethereum;
+    const provider = new ethers.providers.Web3Provider(ethereum);
+    if (!ethereum) {
+      alert("Please install MetaMask to use this dApp!");
+    }
+
+    (async function () {
+      await ethereum.request({ method: "eth_requestAccounts" });
+      const pcd = await SemaphoreSignaturePCDPackage.deserialize(parsed.pcd);
+
+      const msgHash = Buffer.from(
+        ethers.utils.hashMessage(pcd.claim.identityCommitment).slice(2),
+        "hex"
+      );
+      const signatureOfIdentityCommitment = await provider
+        .getSigner()
+        .signMessage(pcd.claim.identityCommitment);
+
+      const poseidon = new Poseidon();
+      await poseidon.initWasm();
+      const treeDepth = 20; // Provided circuits have tree depth = 20
+      const pubKeyTree = new Tree(treeDepth, poseidon);
+
+      // Add some public keys to the tree
+      for (const member of [
+        "0x04b4d5188949bf70c4db5e965a9ea67b80407e8ee7fa3a260ccf86e9c0395fe82cba155fdff55829b3c862322aba402d00b563861b603879ee8ae211c34257d4ad",
+        "0x042d21e6aa2021a991a82d08591fa0528d0bebe4ac9a34d851a74507327d930dec217380bd602fe48a143bb21106ab274d6a51aff396f0e4f7e1e3a8a673d46d83",
+      ]) {
+        pubKeyTree.insert(
+          poseidon.hashPubKey(Buffer.from(member.slice(hexPubkeyOffset), "hex"))
+        );
+      }
+      // Add the prover's public key to the tree
+      const proverPubkeyBuffer: Buffer = Buffer.from(
+        ethers.utils
+          .recoverPublicKey(msgHash, signatureOfIdentityCommitment)
+          .slice(hexPubkeyOffset),
+        "hex"
+      );
+      pubKeyTree.insert(poseidon.hashPubKey(proverPubkeyBuffer));
+      const pubKeyIndex = pubKeyTree.indexOf(
+        poseidon.hashPubKey(proverPubkeyBuffer)
+      ); // == 2 in this test
+
+      // Prove membership of the prover's public key in the tree
+      const merkleProof = pubKeyTree.createProof(pubKeyIndex);
+      const prover = new MembershipProver(pubkeyMembershipConfig);
+      await prover.initWasm();
+
+      const popupUrl = window.location.origin + "/popup";
+      const proofUrl = constructPassportPcdProveAndAddRequestUrl<
+        typeof EthereumGroupPCDPackage
+      >(PASSPORT_URL, popupUrl, EthereumGroupPCDPackage.name, {
+        identity: {
+          argumentType: ArgumentTypeName.PCD,
+          pcdType: SemaphoreIdentityPCDPackage.name,
+          value: undefined,
+          userProvided: true,
+          description:
+            "The Semaphore Identity which you are signing the message.",
+        },
+        groupType: {
+          argumentType: ArgumentTypeName.String,
+          value: GroupType.PUBLICKEY,
+        },
+        signatureOfIdentityCommitment: {
+          argumentType: ArgumentTypeName.String,
+          value: signatureOfIdentityCommitment,
+        },
+        merkleProof: {
+          argumentType: ArgumentTypeName.String,
+          value: JSONBig({ useNativeBigInt: true }).stringify(merkleProof),
```

Comment:
I initially tried to pass the ecdsa-group proof into the `EthereumGroupPCDArgs` but it was too large to fit in the encoded url. Fortunately, the signature + merkleProof fits in the url, but I'm guessing this wouldn't always be the case.